src/plugins/intel_gpu/src/kernel_selector/cl_kernels/gather_ref.cl

```diff
@@ -56,6 +56,9 @@ KERNEL(gather_ref)(
 
     INPUT0_TYPE val = dictionary[dictionary_idx];
 
+    if (dictionary_idx >= DICTIONARY_SIZE)
+        val = 0;
+
```

Comment:
What do you think of using a short form as below?

```suggestion
    INPUT0_TYPE val = (dictionary_idx < DICTIONARY_SIZE) ? dictionary[dictionary_idx] : 0;
```

Reply:
fixed